zeppelin-server/src/main/java/org/apache/zeppelin/rest/InterpreterRestApi.java

```diff
@@ -150,10 +151,15 @@ public Response removeSetting(@PathParam("settingId") String settingId) throws I
   @PUT
   @Path("setting/restart/{settingId}")
   @ZeppelinApi
-  public Response restartSetting(@PathParam("settingId") String settingId) {
-    logger.info("Restart interpreterSetting {}", settingId);
+  public Response restartSetting(String message, @PathParam("settingId") String settingId) {
+    logger.info("Restart interpreterSetting {}, msg={}", settingId, message);
+
     try {
-      interpreterFactory.restart(settingId);
+      RestartInterpreterRequest request = gson.fromJson(message, RestartInterpreterRequest.class);
+
+      String noteId = request == null ? null : request.getNoteId();
```

Comment:
But, I don't think it's not a good implementation because it doesn't look intended. Can you tell me the case that don't have `noteId`?